publisher-sdk/src/main/java/com/criteo/publisher/EpochClock.java

```diff
@@ -18,8 +18,16 @@
 
 public class EpochClock implements Clock {
 
+  private final long startingTime = getCurrentTimeInMillis();
+
   @Override
   public long getCurrentTimeInMillis() {
     return System.currentTimeMillis();
   }
+
+  @Override
+  public long getApplicationTimeInMillis() {
```

Comment:
For the `Thread.sleep`, please see my response in the test file.

Don't you think it might be overkill to have a new class just for this new method ?

Reply:
I don't think it is overkill. I think it is a  completely separate responsibility. A `Clock` is just a clock, it gives current time. Having a `SessionDuration` object that takes a `Clock` as a dependency allows mocking the clock in test (avoiding sleeping).

Reply:
Done